Libraries/StyleSheet/ColorPropType.js

```diff
@@ -9,26 +9,31 @@
   * @providesModule ColorPropType
   */
 'use strict';
+
 var ReactPropTypes = require('ReactPropTypes');
+
 var tinycolor = require('tinycolor');
 
-var colorValidator = function (props, propName) {
+var ColorPropType = function(props, propName) {
   var selectedColor = props[propName];
-  if (selectedColor === null || selectedColor === undefined || selectedColor.toString().trim() === '') {
-    return new Error(
-      `Invalid argument supplied to ${propName}.Expected a string like #123ADF or 'red'.`
-    );
+  if (selectedColor === undefined) {
+    return;
   }
 
-  if (tinycolor(selectedColor.toString().trim()).isValid()) {
-    return null;
+  if (tinycolor(selectedColor) === null) {
+    return new Error(
+`Invalid color supplied to ${propName}: ${selectedColor}. Valid color formats are
+  - '#f0f' (#rgb)
+  - '#ff00ff' (#rrggbb)
+  - '#ff00ff00' (#aarrggbb)
```

Comment:
This should be #rrggbbaa per CSS4: https://drafts.csswg.org/css-color-4/#hex-notation. Let's issue a breaking change.